File: app/src/main/java/com/flutter_android/MyApplication.java
package com.flutter_android;

import android.app.Application;

import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.embedding.engine.FlutterEngineCache;
import io.flutter.embedding.engine.dart.DartExecutor;

public class MyApplication extends Application {
    FlutterEngine flutterEngine;

    @Override
    public void onCreate() {
        super.onCreate();

        // Instantiate a FlutterEngine.
        flutterEngine = new FlutterEngine(this);
        // Configure an initial route.
        flutterEngine.getNavigationChannel().setInitialRoute("/");
        // Start executing Dart code to pre-warm the FlutterEngine.
        flutterEngine.getDartExecutor().executeDartEntrypoint(
                DartExecutor.DartEntrypoint.createDefault()
        );
        // Cache the FlutterEngine to be used by FlutterActivity or FlutterFragment.
        FlutterEngineCache
                .getInstance()
                .put("/",flutterEngine)
                .put("my_engine_id", flutterEngine);
    }
}

File: app/src/main/java/com/flutter_android/MyApplication.kt
package com.flutter_android

import android.app.Application
import io.flutter.embedding.engine.FlutterEngineGroup

class MyApplication : Application() {
    companion object {
        lateinit var context: Application;
    }


    //flutter 引擎组
    lateinit var flutterEngineGroup: FlutterEngineGroup

    override fun onCreate() {
        super.onCreate()
        context = this;
        flutterEngineGroup = FlutterEngineGroup(this)
    }
}